testers/testers/py/markus_python_tester.py

```diff
@@ -141,39 +143,33 @@ def _run_pytest_tests(self, test_file):
         with open(os.devnull, 'w') as null_out:
             try:
                 sys.stdout = null_out
-                verbosity=self.specs.get('pytest_tb_format', 'short')
+                verbosity=self.specs.get('test_data', 'output_verbosity', default='short')
                 with tempfile.NamedTemporaryFile(mode="w+", dir=this_dir) as sf:
                     pytest.main([test_file, f'--junitxml={sf.name}', f'--tb={verbosity}'], plugins=[AddDocstringToJunitXMLPlugin()])
                     results = list(self._parse_junitxml(sf))
             finally:
                 sys.stdout = sys.__stdout__
         return results
 
-
     def run_python_tests(self):
         """
         Return a dict mapping each filename to its results
         """
         results = {}
-        for group in self.specs['runnable_group']:
-            test_file = group.get('script_file_path')
-            if self.specs.get('tester', 'pytest') == 'unittest':
+        for test_file in self.specs.get('test_data', 'script_files', default=[]):
+            if self.specs.get('test_data', 'tester') == 'unittest':
                 result = self._run_unittest_tests(test_file)
             else:
                 result = self._run_pytest_tests(test_file)
             results[test_file] = result
         return results
 
+    @MarkusTester.run_decorator
     def run(self):
-        try:
-            results = self.run_python_tests()
-            with contextlib.ExitStack() as stack:
-                self.feedback_open = (stack.enter_context(open(self.specs['feedback_file'], 'w'))
-                                      if self.specs.get('feedback_file') is not None
-                                      else None)
-                for test_file, result in results.items():
-                    for res in result:
-                        test = self.test_class(self, test_file, res, self.feedback_open)
-                        print(test.run(), flush=True)
-        except Exception as e:
-            print(MarkusTester.error_all(message=str(e)), flush=True)
+        results = self.run_python_tests()
+        feedback_file = self.specs.get('test_data', 'feedback_file_name')
+        with MarkusTester.open_feedback(feedback_file) as feedback_open:
```

Comment:
Does not belong in this pr, but I think open_feedback can be smarter and save us a bit of typing, if the 'feedback_file_name' setting is always under 'test_data'.
It could be an instance method, using self.specs to fetch the feedback file name itself: `with self.open_feedback as feedback_open:`

Reply:
Agreed, but let's do it in a separate PR once all the PRs for the individual testers have been pulled in so that we can make the change all at once.